pkg/common/endpoint.go

```diff
@@ -129,32 +156,32 @@ func (ep *RetinaEndpoint) SetAnnotations(annotations map[string]string) {
 	}
 }
 
-func (e *RetinaEndpoint) PrimaryIP() (string, error) {
-	e.RLock()
-	defer e.RUnlock()
+func (ep *RetinaEndpoint) PrimaryIP() (string, error) {
+	ep.RLock()
+	defer ep.RUnlock()
 
-	if e.ips != nil {
-		pip := e.ips.PrimaryIP()
+	if ep.ips != nil {
+		pip := ep.ips.PrimaryIP()
 		if pip != "" {
 			return pip, nil
 		}
 	}
 
-	return "", fmt.Errorf("no primary IP found for endpoint %s", e.Key())
+	return "", noPrimaryIPFoundForEndpointError(ep.Key())
```

Comment:
incidentally, if we wrap errors, we can do that here instead of needing a helper func